src/main/groovy/com/calgaryscientific/gradle/VeracodeWorkflowSandboxTask.groovy

```diff
@@ -0,0 +1,125 @@
+    String waitTimeBetweenAttempts
+    String delete
+
+    VeracodeWorkflowSandboxTask() {
+        group = 'Veracode Sandbox'
+        description = "Run through the Veracode Workflow for the given 'app_id' and 'sandbox_id' using 'build_version' as the build identifier"
+        requiredArguments << 'app_id' << 'sandbox_id' << 'build_version'
+        optionalArguments << 'maxUploadAttempts' << 'waitTimeBetweenAttempts' << 'delete'
+        app_id = project.findProperty("app_id")
+        sandbox_id = project.findProperty("sandbox_id")
+        build_version = project.findProperty("build_version")
+    }
+
+    VeracodeGetBuildInfoSandboxTask buildInfoSandboxTask = new VeracodeGetBuildInfoSandboxTask()
+    VeracodeCreateBuildSandboxTask createBuildSandboxTask = new VeracodeCreateBuildSandboxTask()
+    VeracodeUploadFileSandboxTask uploadFileSandboxTask = new VeracodeUploadFileSandboxTask()
+    VeracodeBeginPreScanSandboxTask beginPreScanSandboxTask = new VeracodeBeginPreScanSandboxTask()
+    VeracodeBeginScanSandboxTask beginScanSandboxTask = new VeracodeBeginScanSandboxTask()
+
+    void run() {
+        // Retrieve the properties in the run method to allow for testing
+        // TODO: Look for a better way
+        app_id = project.findProperty("app_id")
+        sandbox_id = project.findProperty("sandbox_id")
+        build_version = project.findProperty("build_version")
+        maxUploadAttempts = project.findProperty("maxUploadAttempts")
+        waitTimeBetweenAttempts = project.findProperty("waitTimeBetweenAttempts")
+        delete = project.findProperty("delete")
+        println "[debug] app_id: ${app_id}"
+        println "[debug] sandbox_id: ${sandbox_id}"
+        println "[debug] build_version: ${build_version}"
+
+        // Apply default properties to tasks
+        buildInfoSandboxTask.app_id = app_id
+        buildInfoSandboxTask.sandbox_id = sandbox_id
+        buildInfoSandboxTask.veracodeAPI = veracodeAPI
+
+        createBuildSandboxTask.app_id = app_id
+        createBuildSandboxTask.sandbox_id = sandbox_id
+        createBuildSandboxTask.build_version = build_version
+        createBuildSandboxTask.veracodeAPI = veracodeAPI
+
+        uploadFileSandboxTask.app_id = app_id
+        uploadFileSandboxTask.sandbox_id = sandbox_id
+        uploadFileSandboxTask.maxUploadAttempts = maxUploadAttempts
+        uploadFileSandboxTask.waitTimeBetweenAttempts = waitTimeBetweenAttempts
+        uploadFileSandboxTask.delete = delete
+        uploadFileSandboxTask.veracodeAPI = veracodeAPI
+
+        beginPreScanSandboxTask.app_id = app_id
+        beginPreScanSandboxTask.sandbox_id = sandbox_id
+        beginPreScanSandboxTask.veracodeAPI = veracodeAPI
+
+        beginScanSandboxTask.app_id = app_id
+        beginScanSandboxTask.sandbox_id = sandbox_id
+        beginScanSandboxTask.veracodeAPI = veracodeAPI
+
+        // Get build info
+        printf "[debug] getBuildInfo\n"
+        buildInfoSandboxTask.run()
```

Comment:
I'm not a big fan of field/property injection as it's not explicit what's require to use the object.
- couple alternatives I could see for buildInfoSandboxTask.run()
  - new GetBuildInfoSandbox(sandbox_id, app_id).getBuildInfo()
  - buildInfoSandbox.getBuildInfo(sandbox_id, app_id)

Reply:
I can't call the constructor within a method of a gradle task. It has to be as a global so gradle can figure our dependencies and such. That is why I couldn't do something like that.

Also, gradle plugins depend on external properties passed through the command line or defined on the gradle.properties. Injecting the properties seems to me like it aligns with what I have been doing so far across the entire plugin.

Reply:
"can't call constructor" within a method is a smell for me -> it sounds like we depends on gradle on how it figures out dependencies and properties so much that we can't call constructor in the method. Again, if you extract domain logic away from gradle custom task, you would have more flexibility here.....